src/steps/navigationSteps.ts

```diff
@@ -0,0 +1,121 @@
+import { config } from './../config';
+import {getNavigationState, getStepsFromNavigation, steps } from './steps';
+
+
+// We only report the active steps in the last 2 navigation step to prevent
+// Convert the steps.navigationSteps to an array of active steps.
+// "stale steps" from being reported.
+//
+// "stale steps" are defined as steps where the start mark was executed but the
+// corresponding end mark was never executed - thus leaving it as a "stale step"
+export const getActiveStepsFromNavigationSteps = (): Record<string, boolean>  => {
+  const navigationSteps = getNavigationState();
+  const { startMarkToStepsMap } = steps;
+  if (Object.keys(navigationSteps).length === 0) {
```

Comment:
You call `Object.keys(navigationSteps)` twice in the same function, why not just store the value? It will make this function faster.

Reply:
Good callout, will update